middleware/src/middleware.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+const request = require('request');
+const url = require('url');
+
+/**
+ * A default set of user agent patterns for bots/crawlers that do not perform
+ * well with pages that require JavaScript.
+ */
+const botUserAgents = module.exports.botUserAgents = [
+  'W3C_Validator',
+  'baiduspider',
+  'bingbot',
+  'embedly',
+  'facebookexternalhit',
+  'linkedinbo',
+  'outbrain',
+  'pinterest',
+  'quora link preview',
+  'rogerbo',
+  'showyoubot',
+  'slackbot',
+  'twitterbot',
+  'vkShare',
```

Comment:
Do you think we should pull this out and publish it separately? Either as a JSON file, or node module.